src/configure/clients/ProvisioningServiceClient.ts

```diff
@@ -0,0 +1,58 @@
+import { ServiceClientCredentials, UrlBasedRequestPrepareOptions } from "ms-rest";
+import { IServiceUrlDefinition, ServiceFramework } from "../helper/remoteServiceUrlHelper";
+import { ProvisioningConfiguration } from "../model/provisioningConfiguration";
+import { IProvisioningServiceClient } from "./IProvisioningServiceClient";
+import { RestClient } from "./restClient";
+
+export class ProvisioningServiceClient implements IProvisioningServiceClient {
+  private restClient: RestClient;
+  private serviceDefinition: IServiceUrlDefinition;
+  private defaultHeaders: { [propertyName: string]: string };
+  private readonly pipelineProvisioningJob = "PipelineProvisioningJob";
+  private readonly PEProvisioningServiceAPIVersion = "6.1-preview.1";
+
+  constructor( serviceDefinition: IServiceUrlDefinition, headers: { [propertyName: string]: string }, credentials: ServiceClientCredentials) {
+    this.restClient = new RestClient(credentials);
+    this.serviceDefinition = serviceDefinition;
+    this.defaultHeaders = headers;
+  }
+
+  public async createProvisioningConfiguration(provisioningConfiguration: ProvisioningConfiguration, githubOrg: string, repositoryId: string): Promise<ProvisioningConfiguration> {
+    const requestUrl = this.serviceDefinition.serviceUrl + githubOrg + "/" + repositoryId + "/" + this.pipelineProvisioningJob;
+    // tslint:disable-next-line:prefer-const
+    let queryParams: { [propertyName: string]: string };
```

Comment:
you can create default query params in constructor and use it here and in other function